src/base/QXmppStun.cpp

```diff
@@ -2350,11 +2350,11 @@ static QList<QUdpSocket*> reservePort(const QList<QHostAddress> &addresses, quin
 {
     QList<QUdpSocket*> sockets;
     foreach (const QHostAddress &address, addresses) {
-        QUdpSocket *socket = new QUdpSocket(parent);
+        auto *socket = new QUdpSocket(parent);
         sockets << socket;
         if (!socket->bind(address, port)) {
-            for (int i = 0; i < sockets.size(); ++i)
-                delete sockets[i];
+            for (auto &socket : sockets)
+                delete socket;
```

Comment:
`qDeleteAll(sockets);`

Reply:
Personally I'd prefer using `std::unique_ptr` or `std::shared_ptr` for this.

Reply:
See below